common/stream.h

```diff
@@ -273,19 +291,19 @@ inline stream_in_t::stream_in_t(const std::vector<uint8_t>& buffer) :
 template<typename TType>
 inline void stream_in_t::pop(TType& value)
 {
-	if constexpr (std::is_trivial_v<TType> &&
+	if constexpr (std::is_trivially_copyable_v<TType> &&
 	              !(std::is_pointer_v<TType> ||
 	                std::is_reference_v<TType>))
 	{
 		if (inBuffer.size() - inPosition < sizeof(TType))
 		{
-			memset(&value, 0, sizeof(TType));
+			value = TType{};
 			inPosition = this->inBuffer.size();
 			failed = true;
 			return;
 		}
 
-		memcpy(&value, &inBuffer[inPosition], sizeof(TType));
+		value = reinterpret_cast<const TType&>(inBuffer[inPosition]);
```

Comment:
Why is `memcpy` deemed worse than using `reinterpret_cast` to a reference for copying trivially copyable objects?

In such context both `memcpy` and `reinterpret_cast` should be of similar performance, compiler will translate them both to an operation like `mov`.

IMHO `std::memcpy` is more idiomatic for copying bytes from a buffer to a target variable in this scenario.

Also I would use `std::memcpy(&value, inBuffer.data() + inPosition, sizeof(TType));` instead of `&inBuffer[inPosition]` for a second argument

Reply:
Our intent is not to copy some memory from one place to another, rather to treat position in buffer as `TType`.
Using `reinterpret_cast` reduces "byte magic" to a minimum, and states the change of type explicitly.